File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/service/SignatureService.java
package com.shterneregen.securelan.crypto.service;

import java.security.PrivateKey;
import java.security.PublicKey;

public interface SignatureService {
    byte[] sign(byte[] data, PrivateKey privateKey);

    boolean verify(byte[] data, byte[] signature, PublicKey publicKey);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/service/SignatureService.kt
package com.shterneregen.securelan.crypto.service

import java.security.PrivateKey
import java.security.PublicKey

interface SignatureService {
    fun sign(data: ByteArray, privateKey: PrivateKey): ByteArray

    fun verify(data: ByteArray, signature: ByteArray, publicKey: PublicKey): Boolean
}
